frontend/packages/ceph-storage-plugin/src/features.ts

```diff
@@ -70,34 +69,30 @@ export const detectRGW: FeatureDetector = async (dispatch) => {
 
 export const detectOCS: FeatureDetector = async (dispatch) => {
   try {
-    const storageCluster = await fetchK8s(
-      OCSServiceModel,
-      OCS_INTERNAL_CR_NAME,
-      CEPH_STORAGE_NAMESPACE,
+    const storageClusters = await k8sList(OCSServiceModel, { ns: CEPH_STORAGE_NAMESPACE });
+    const storageCluster = storageClusters.find(
+      (sc: OCSStorageClusterKind) => sc.status.phase !== 'Ignored',
     );
+    const isInternal = _.isEmpty(storageCluster.spec.externalStorage);
```

Comment:
What if I make it `false` ? 
Should not this check be based on the value for `spec.externalStorage.enable`  ?
```
        externalStorage: {
          enable: false,
        },
```

Reply:
In external mode, we set it to false. For every other mode, it's an empty object. Never is there a situation where `externalStorage.enable` is `false`. You can't flip a storage cluster to external/internal by just changing that field.

Reply:
Checking over spec.externalStorage.enable would be still a better option IMO.
It will scale well wrt any changes in future.
Right now UI is assumed to be not sending this field for internal and attached devices, operator by default makes it {} .
If any change happen we need to track to make the fix here as well.